### ISSUE TITLE
Multi module streams configuration

### ISSUE THREAD

**Issue by gdonoso94** [spring-cloud/spring-cloud-stream#2972]
Hello,

I'm not sure this should be considered a bug, it's more a _how-to_ question.

I have a multi module project, consisting of multiple gradle modules where every module create different consumers and producers using a RabbitMQ broker binding. The consumer functions, bindings and binders are defined in the _global_ application.yaml file.

In an attempt of structure better the project, I've split the config in separate application.yaml files, one under each module resources', only containing the needed module related configuration.

When there are multiple application.yaml files, these are _combined_ and, since "spring.cloud.stream.function.definition" is a string, it get's overwritten by the different application.yaml files, therefore making impossible to split the configuration files in the project.

Diving a little bit in this repo, I've seen the defintions are obtained and split [here](https://github.com/spring-cloud/spring-cloud-stream/blob/b5eccf623240a497f06c4ea5f3e20fa6fa883182/core/spring-cloud-stream/src/main/java/org/springframework/cloud/stream/function/FunctionConfiguration.java#L1010), and I was wondering if it could be possible to have the definitions declared as an array in the config, so it can be extended when having multiple yaml configuration files. In my projects I've used @ConfigurationProperties for this and it's something I believe it could help here.

I would be more than happy to help with a PR for this but I need some guidance, this would be the first open source project I collaborate with and I'm pretty sure I'm missing something.

**Comment by gdonoso94** [comment]
Any thoughts? 🤔

**Comment by olegz** [comment]
I personally have no tried it nor have we had any requests like this.
You can define multiple function definitions in a single property by delimiting them by semicolon (i.e., `s.c.s.function.definition=foo;bar;baz`), but I am not sure that is what you are looking for.
Perhaps contributing and submitting a PR would be better so we can review and understand better what you are trying to accomplish.

**Comment by gdonoso94** [comment]
Hi Oleg, thanks for your response

I know I can define multiple function definitions. The problem arises when multiple configuration `yaml` define those. The yaml are combined and the s.c.s.function.definition gets overriden...

**Comment by olegz** [comment]
@gdonoso94 I am having hard time reproducing it. In fact I see a different behavior then you describe.

So, here is what I have:
*Module A:*
- I have `MyConfiguration` class which has `name` and `'id` properties
- I have `application.properties` defining values for `name` property - "Bob"

*Module B:*
 - I have `application.properties` defining values for `id` property - "123"

*Module B* is a dependency to *Module A* and when i run *Module A* `MyConfiguration` gets initialised only with `name` property, but no `id`.
Now. . . if I remove `application.properties` from *Module A* and run *Module A* again, `MyConfiguration` gets initialised only with `id` property, but no `name`.

In other words, i don't see an override, rather one configuration or the other gets picked up.

Perhaps my setup does not exactly re[resent yours, so may be you can create a small reproducible sample that I can play with.

**Comment by gdonoso94** [comment]
Hi @olegz , thanks for your response. I will try to provide some minimal example as soon as I find some spare time.

**Comment by omercelikceng** [comment]
@olegz  I understand the issue, and I have previously encountered it. If you allow me, I would like to work on resolving this problem in my spare time.

**Comment by olegz** [comment]
@gdonoso94 Any updates?
